docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,181 @@
+
+> Any AWS service or action that generates S3 object events may trigger the EDP service.
+
+## Requirements
+
+1. A pre-existing VPC that includes a [NAT
+Gateway](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html)
+or the following [VPC
+endpoints](https://docs.aws.amazon.com/vpc/latest/privatelink/concepts.html#concepts-vpc-endpoints):
+  - Amazon S3
+  ([gateway](https://docs.aws.amazon.com/vpc/latest/privatelink/gateway-endpoints.html)
+  or
+  [interface](https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html))
+  - EventBridge (interface)
+2. [Enable EventBridge S3
+Events](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-event-notifications-eventbridge.html)
+for all S3 buckets to be monitored by EDP
+
+![](../imgs/edp-event-bridge-s3-properties.png)
+
+## Deployment
+
+To use EDP in your Quilt instance, you will need an additional
+CloudFormation template to configure when, and how, events get
+generated.
+
+### Parameters
+
+The following parameters need to be defined in the CloudFormation
+template during deployment:
+
+- `VPC`: Same as the existing Quilt instance.
+- `SecurityGroup`: Same as the existing Quilt instance.
+- `Subnets`: Same two subnets as the existing Quilt instance.
+- `BucketName`: Name of the Amazon S3 bucket to monitor. Must be
+read accessible to EDP.
+- `BucketIgnorePrefixes`: List of bucket path segments to ignore
+- `BucketPrefixDepth`: The number of `/`-separated path segments
+at the beginning of an S3 object key. Default value is `2`.
+- `BucketThresholdDuration`: Trigger a notification when this number
+of seconds has elapsed since the last object event in the S3 bucket 
+occurred. Default value is `300` seconds.
+- `BucketThresholdEventCount`: Trigger a notification when this
+number of files have been created (since the prior trigger)
+- `DBUser`: Username for EDP RDS instance
+- `DBPassword`: Password for EDP RDS instance
+- `EventBusName`: Name of custom EventBridge event bus that recieves events
+
+![](../imgs/edp-cloudformation-parameters.png)
+
+## How EDP works
+
+1. EDP monitors S3 object events for _s3://source-bucket_
+2. After a fixed number of events (`BucketThresholdEventCount`) or
+a maximum duration (`BucketThresholdDuration`) EDP creates a
+package in _s3://target-bucket_ (the _target_ bucket may or may not be
+the same as the _source_ bucket).
+3. Each EDP `files_ready` event contains sufficient information for the
+recipient to make Quilt data packages from the event:
+  - S3 Bucket name
+  - Common prefix
+  - Number of files
+  - Timestamp of event
+4. EDP ignores changes to ignored prefixes (`BucketIgnorePrefixes` parameter)
+in the source S3 bucket (e.g. `raw/*`)
+5. EDP publishes events to an AWS EventBridge bus and can forward
+events to external EventBridge buses
+6. Users subscribe the Amazon SNS topic created to recieve notifications
+
+## Example EDP events
+
+Currently, EDP generates two different `files_ready` events:
```

Comment:
This section and the above are super confusing and wrong. We say `package-data-ready` here then the example shows `"package-objects-ready"` and above say `files_ready`. Need to consolidate terms.

Reply:
Hmmm. We say:
 91 Currently, EDP generates two different `files_ready` events:
 92 - `package-data-ready`
 93 - `package-ready`

These are the two types of event. It's explicit.

Reply:
@sir-sigurd We call the `detail type``package-objects-ready` but is it actually `package-data-ready`? There's no other mention of `package-objects-ready` in the docs.

Reply:
files_ready is a made-up "type" AFAICT. Those are simply EDP events unless we have an actual "files_ready" type.

Reply:
`package-data-ready` was renamed to `package-objects-ready` at some point. `files_ready` is a made-up "type" and I think we should not use it to avoid confusion.